scripts/run/update.sh

```diff
@@ -1,78 +1,241 @@
 #!/bin/bash
+#
+# This script automates the process of updating the FiberPTS system. It ensures
+# network connectivity, pulls the latest changes for a specified branch from the
+# repository, updates service files based on the latest templates, restarts services,
+# and checks the system status to confirm successful updates. If necessary, it can
+# also rollback changes in case of update failure.
 
-SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-SETUP_DIR="${SCRIPT_DIR}/../setup"
+CWD="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly CWD
 
-function assert_root() {
-  # Root check
-  if [ "$(id -u)" -ne 0 ]; then
-    echo "This script must be run as root. Please use sudo."
-    exit 1
-  fi
-}
+set +a
+source "${CWD}"/../globals.sh || return 1
+source "${CWD}"/../paths.sh || return 1
+set -a
 
-function load_env_variables() {
-  set -a
-  source "${SCRIPT_DIR}/../../scripts/paths.sh" || return 1
-  source "${SCRIPT_DIR}/../../.env" || return 1
-  set +a
+#######################################
+# Connects to a WiFi network using NetworkManager CLI (nmcli).
+# Globals:
+#   None
+# Arguments:
+#   None
+# Outputs:
+#   Prompts user for SSID and password. Outputs nmcli command results.
+#######################################
+function connect_network() {
+  declare ssid pwd
+
+  read -rp "SSID: " ssid
+  read -rsp "Password: " pwd
+  nmcli device wifi connect "${ssid}" password "${pwd}"
+  return $?
 }
 
-# TODO: Test if this works since may need to do a hard reset first
+#######################################
+# Pulls the latest changes from a specified Git branch.
+# Globals:
+#   FAIL, RED, RESET - Used for coloring the error message.
+# Arguments:
+#   $1 - The name of the branch to pull changes from.
+# Outputs:
+#   Status messages regarding the pull operation.
+#######################################
 function pull_latest_changes() {
-  echo "Pulling latest changes from ${GIT_BRANCH} branch..."
-  git -C "${PROJECT_PATH}" pull origin "${GIT_BRANCH}" || {
-    echo "Git pull failed" >&2
+  echo "Pulling latest changes from $1..."
+
+  if ! git pull origin "$1"; then
```

Comment:
Does this not require you to be in the `/opt/FiberPTS` directory?

Reply:
Nope. This assumes that we are running our scripts when our `CWD` is `/opt/FiberPTS`.

Reply:
Is there a check to ensure the `CWD` is `/opt/FiberPTS` when running the script?

Reply:
Done.